src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java

```diff
@@ -1162,26 +1162,36 @@ private List<SupplierAttachmentResponse> getSupplierAttachmentResponses(final In
     return supplierAttachmentResponsesList;
   }
 
-  private List<Offer> getOffersWithSupplierAttachments(final ExportRfxResponse exportRfxResponse,
-      final String supplierId) {
+  private List<Offer> getOffersWithSupplierAttachments(
+      final ExportRfxResponse exportRfxResponse, final String supplierId) {
     List<Offer> offersWithParameters;
 
     if (Objects.isNull(supplierId)) {
-      offersWithParameters = exportRfxResponse.getOffersList().getOffer().stream()
-          .filter(offer -> ObjectUtils.allNotNull(offer, offer.getTechOffer(),
-              offer.getTechOffer().getParameterResponses(),
-              offer.getTechOffer().getParameterResponses().getParameter())
-              && !offer.getTechOffer().getParameterResponses().getParameter().isEmpty())
-          .collect(Collectors.toList());
+      offersWithParameters =
+          exportRfxResponse.getOffersList().getOffer().stream()
+              .filter(
+                  offer ->
+                      ((null != offer)
```

Comment:
Does that ObjectUtils.allNull do this a bit more cleanly?

Reply:
ObjectUtils.allNotNull , or allNull works well different objects as it loops through one by one.
But if we are going though object graph then, it doesn't avoid calling null.getXXXXX(), which is causing issue here. 



```
 Object[] var1 = values;
            int var2 = values.length;

            for(int var3 = 0; var3 < var2; ++var3) {
                Object val = var1[var3];
                if (val == null) {
                    return false;
                }
            }
```